.github/workflows/wheels-build.yml

```diff
@@ -123,6 +114,52 @@ jobs:
         with:
           path: ./wheelhouse/*.whl
           name: ${{ inputs.artifact-prefix }}wheels-tier-1-${{ matrix.os }}
+  wheels-tier-1-aarch64:
+    name: "Wheels / Tier 1"
+    if: (inputs.wheels-tier-1 == 'default' && inputs.default-action || inputs.wheels-tier-1) == 'build'
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-24.04-arm
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: ${{ inputs.python-version }}
+      - uses: dtolnay/rust-toolchain@stable
+        with:
+          components: llvm-tools-preview
+      - name: Configure PGO
+        shell: bash
+        if: inputs.pgo
+        # The `$GITHUB_ENV` magic file uses some sort of custom parsing, so the variables shouldn't
+        # be quoted like they would be if bash were interpreting them.  You still need to use quotes
+        # to avoid word splitting where appropriate in compound environment variables.
+        #
+        # Beware that the heredoc is configured to expand bash variables, but cibuildwheel has
+        # special handling for certain variables (`$PATH`, in particular), so you may need to escape
+        # some dollar signs to pass those through to cibuildwheel as variables, or use single quotes
+        # to prevent shell expansion.
+        run: |
+          set -e
+          mkdir -p "$PGO_WORK_DIR"
+
+          cat >>"$GITHUB_ENV" <<EOF
+          CIBW_BEFORE_ALL=yum install -y wget && {package}/tools/install_rust_msrv.sh
+          CIBW_BEFORE_BUILD=bash ./tools/build_pgo.sh $PGO_WORK_DIR $PGO_OUT_PATH
+          CIBW_ENVIRONMENT_LINUX=RUSTFLAGS='-Cprofile-use=$PGO_OUT_PATH -Cllvm-args=-pgo-warn-missing-function' PATH="\$PATH:\$HOME/.cargo/bin" CARGO_NET_GIT_FETCH_WITH_CLI="true"
+          EOF
+        env:
+          PGO_WORK_DIR: ${{ github.workspace }}/pgo-data
+          PGO_OUT_PATH: ${{ github.workspace }}/merged.profdata
```

Comment:
I'm fine to live this in a separate job for the initial support just to get it merged after I know how much of a slog it just was, but I might like to follow up with a little bit of logic that moves the AArch64 config into the regular Tier 1 matrix. Now that we've got the working config, and the difference seems to be only in the `CIBW_BEFORE_ALL` and not setting `RUSTUP_TOOLCHAIN` (for the env config, I mean), I think it's possible to de-duplicate the job a bit more, which is just a nice-to-have - we've definitely missed updating configs in the past in the CI jobs when we had them duplicated (on Aer more than Terra, but still).

Reply:
I've deduplicated the job to do the toolchain switching logic in the run step (assuming I got the logic and syntax correct in the current iteration)